packit/config/common_package_config.py

```diff
@@ -252,55 +304,13 @@ def files_to_sync(self) -> List[SyncFilesItem]:
-        return (
-            "CommonPackageConfig("
-            f"specfile_path='{self.specfile_path}', "
-            f"files_to_sync='{self.files_to_sync}', "
-            f"dist_git_namespace='{self.dist_git_namespace}', "
-            f"upstream_project_url='{self.upstream_project_url}', "
-            f"upstream_package_name='{self.upstream_package_name}', "
-            f"downstream_project_url='{self.downstream_project_url}', "
-            f"downstream_package_name='{self.downstream_package_name}', "
-            f"dist_git_base_url='{self.dist_git_base_url}', "
-            f"actions='{self.actions}', "
-            f"upstream_ref='{self.upstream_ref}', "
-            f"allowed_gpg_keys='{self.allowed_gpg_keys}', "
-            f"create_pr='{self.create_pr}', "
-            f"create_sync_note='{self.create_sync_note}', "
-            f"spec_source_id='{self.spec_source_id}', "
-            f"upstream_tag_template='{self.upstream_tag_template}', "
-            f"patch_generation_ignore_paths='{self.patch_generation_ignore_paths}',"
-            f"patch_generation_patch_id_digits='{self.patch_generation_patch_id_digits}',"
-            f"copy_upstream_release_description='{self.copy_upstream_release_description}',"
-            f"sources='{self.sources}', "
-            f"merge_pr_in_ci={self.merge_pr_in_ci}, "
-            f"srpm_build_deps={self.srpm_build_deps}, "
-            f"identifier={self.identifier}, "
-            f"packit_instances={self.packit_instances}, "
-            f"issue_repository='{self.issue_repository}', "
-            f"release_suffix='{self.release_suffix}', "
-            # from deprecated JobMetadataConfig
-            f"targets={self._targets}, "
-            f"timeout={self.timeout}, "
-            f"owner={self.owner}, "
-            f"project={self.project}, "
-            f"dist_git_branches={self.dist_git_branches}, "
-            f"branch={self.branch}, "
-            f"scratch={self.scratch}, "
-            f"list_on_homepage={self.list_on_homepage}, "
-            f"preserve_project={self.preserve_project}, "
-            f"additional_packages={self.additional_packages}, "
-            f"additional_repos={self.additional_repos}, "
-            f"fmf_url={self.fmf_url}, "
-            f"fmf_ref={self.fmf_ref}, "
-            f"use_internal_tf={self.use_internal_tf}, "
-            f"skip_build={self.skip_build},"
-            f"env={self.env},"
-            f"enable_net={self.enable_net},"
-            f"allowed_pr_authors={self.allowed_pr_authors},"
-            f"allowed_committers={self.allowed_commiters},"
-            f"tf_post_install_script='''{self.tf_post_install_script}''')"
-        )
+        # required to avoid cyclical imports
+        from packit.schema import CommonConfigSchema
+
+        s = CommonConfigSchema()
+        # For __repr__() return a JSON-encoded string, by using dumps().
+        # Mind the 's'!
+        return f"CommonPackageConfig: {s.dumps(self)}"
```

Comment:
Nice!

Can we maybe cover this by parenthesis to preserve the init-like format?

I am actually wondering why we don't use data classes to resolve all of this.. (Sorry for off-topic..;)

Reply:
> Nice!
> 
> Can we maybe cover this by parenthesis to preserve the init-like format?

That would look like this, which doesn't really resemble an init-method, and I thought it might be confusing for some:

```
CommonPackageConfig({"patch_generation_patch_id_digits": 4, ... })
```

But I might be wrong...

> 
> I am actually wondering why we don't use data classes to resolve all of this.. (Sorry for off-topic..;)

Because of CentOS Stream 8 and RHEL 8 having Python 3.6, which doesn't have data classes.